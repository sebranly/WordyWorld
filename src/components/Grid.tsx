// Vendor
import * as React from "react";
import { Col, Grid as EasyGrid, Row } from "react-native-easy-grid";
import { StyleSheet, View } from "react-native";
import { Text } from "react-native";

// Internal
import { Word } from "../types/interfaces";
import { GAME_ROWS } from "../config/settings";

export interface GridProps {
  word: Word;
}

const Grid: React.FC<GridProps> = (props) => {
  const { word } = props;

  const renderRow = (indexRow: number, last?: boolean) => {
    const renderColumn = (letter: string, indexCol: number) => {
      const innerComponent = last ? (
        <View style={styles.viewText}>
          <Text style={styles.text}>{letter}</Text>
        </View>
      ) : null;

      return (
        <Col key={`col-${indexCol}`} style={styles.cell}>
          {innerComponent}
        </Col>
      );
    };

    const renderColumns = () => {
      const letters = word.en.split("");

      const columns = letters.map((v, i) => renderColumn(v, i));
      return columns;
    };

    return <Row key={`row-${indexRow}`}>{renderColumns()}</Row>;
  };

  const renderRows = () => {
    const lastIndexRow = GAME_ROWS - 1;
    const allButLastRow = Array.from({ length: lastIndexRow }, (_v, i) =>
      renderRow(i)
    );

    return [...allButLastRow, renderRow(lastIndexRow, true)];
  };

  return <EasyGrid style={styles.grid}>{renderRows()}</EasyGrid>;
};

const styles = StyleSheet.create({
  cell: {
    backgroundColor: "grey",
    margin: 2,
  },
  grid: {
    flex: 3,
  },
  text: { fontSize: 30 },
  viewText: {
    alignItems: "center",
    flex: 1,
    justifyContent: "center",
  },
});

export { Grid };
