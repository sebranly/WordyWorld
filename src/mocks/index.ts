// Internal
import { Word } from "../types/interfaces";

const mockWords: Word[] = [
  { en: "car", fr: "voiture" },
  { en: "cat", fr: "chat" }
];

export { mockWords };
