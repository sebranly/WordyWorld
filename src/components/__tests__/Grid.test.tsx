// Vendor
import React from "react";
import { render } from "@testing-library/react-native";

// Internal
import { Grid } from "../Grid";
import { mockWords } from "../../mocks";

it("renders", () => {
  const props = { word: mockWords[0] };

  const { container } = render(<Grid {...props} />);
  expect(container.children.length).toBeGreaterThan(0);
  expect(container.children).toMatchSnapshot();
});
