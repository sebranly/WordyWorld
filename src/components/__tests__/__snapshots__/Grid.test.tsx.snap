// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <View
    style={
      Object {
        "flex": 3,
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 30,
              }
            }
          >
            c
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 30,
              }
            }
          >
            a
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "grey",
            "flex": 1,
            "flexDirection": "column",
            "margin": 2,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 30,
              }
            }
          >
            r
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;
